tests/test_serialize_response_model.py

```diff
@@ -152,3 +153,32 @@ def test_validdict_exclude_unset():
         "k2": {"aliased_name": "bar", "price": 1.0},
         "k3": {"aliased_name": "baz", "price": 2.0, "owner_ids": [1, 2, 3]},
     }
+
+
+class Toggle(BaseModel):
+    switch: bool
+
+    if PYDANTIC_V2:
+        from pydantic import field_validator
+
+        @field_validator("switch")
+        def toggle_switch(cls, switch: bool):
+            return not switch
+
+    else:
+        from pydantic import validator
+
+        @validator("switch")
+        def toggle_switch(cls, v):
+            return not v
```

Comment:
not sure if there's a better way to test this 😊

@kaiix I know this is an old PR, but feel free to suggest something :D

Reply:
happy to see this PR finally has a chance to be merged

I think its enough to illustrate the issue, any better suggestions?  @patrick91

Reply:
Thanks! Currently, this test passes in `master` :sweat_smile:

We would need a test that doesn't pass in `master` but passes with this PR. :nerd_face:

Reply:
I see, this test worked fine when this PR was created (v0.71.0), years of updates have affected the results.

I'll try to fix the test.